### ISSUE TITLE
Error: adb:preparesystemimage: Error: {"error":{"code":1,"cmd":"adb -P 5037 shell mkdir -p /cache/recovery"},"stderr":"mkdir: '/cache/recovery': Required key not available"}

### ISSUE THREAD

**Issue by strangeli** [ubports/ubports-installer#2155]
**UBports Installer `0.8.9-beta` (deb)**
Environment: `Ubuntu 18.04.6 LTS Bionic Beaver linux 5.4.0-84-generic x64 NodeJS v12.18.3`
Device: FP3
Target OS: Ubuntu Touch
Settings: `undefined`
OPEN-CUTS run: *N/A*
Log: *N/A*

I don't know...

**Comment by luksus42** [comment]
Hi, did you check the 'wipe' option in tje installer?
If not, please do it (it will reformat datapartion and delete possibly existing userdata) and try again.

**Comment by Fuseteam** [comment]
appearent according to https://forum.xda-developers.com/t/im-having-problems-with-twrp-and-orangefox-permission-errors-all-the-time.4203951/ this may be related to an encrypted data partition, perhaps you can boot into recovery and wipe that parition

**Comment by werA111** [comment]
same problem on **OS** Ubuntu 20.04.3 LTS, **Device** FP3
I had /e/ Android 10

**Comment by werA111** [comment]
Yes, I checked the wipe option

**Comment by strangeli** [comment]
I used this one here last time to get Android 10 back on which worked fine (the link from the ubports-installer): 

https://androidfilehost.com/?fid=7161016148664787652

(Still have not tried the wipe option though, will do in the coming days)

**Comment by werA111** [comment]
Thanks @luksus42 , it worked!

**Comment by luksus42** [comment]
@strangeli , is it solved for you too?

**Comment by luksus42** [comment]
> Just to save some time: Since I still have /e/ OS now, I need to flash Android anyway before using TWRP?

Yes, you should install latest fairphone OS (Android 10) before installing Ubuntu Touch. Maybe TWRP is not needed then anymore. Only if you run into the same issues like before, you should try the TWRP wipe way.

**Comment by luksus42** [comment]
This will be solved by:
https://github.com/ubports/installer-configs/pull/152

**Comment by luksus42** [comment]
@strangeli 
>  I just wondered if I can skip the Android installation
No, it's needed to flash some partitions to it's "original" state.
As far as I know its mostly about the vendor partition. But that needs to be tested.

**Comment by NeoTheThird** [comment]
looks like this is still happening in `0.9.3-beta`:

`stack trace: Error: {"error":{"code":1,"cmd":"adb -P 5037 shell mkdir -p /cache/recovery"},"stderr":"mkdir: '/cache/recovery': Required key not available"}`
from https://ubports.open-cuts.org/run/6233865155cdf00008ffa3f9

Am i understanding https://github.com/ubports/installer-configs/pull/152/files correctly that the user simply needs to enable wipe and try again? In that case, we could just add some error handling for `Required key not available` and display an appropriate generic message. Would that make sense?

**Comment by NeoTheThird** [comment]
we could also use a handler for this, see https://github.com/ubports/installer-configs/issues/175

**Comment by luksus42** [comment]
@NeoTheThird 
> Am i understanding https://github.com/ubports/installer-configs/pull/152/files correctly that the user simply needs to enable wipe and try again? In that case, we could just add some error handling for `Required key not available` and display an appropriate generic message. Would that make sense?

It depends on how the installer-config is implementing the "wipe" option. In the case of the FP3, a simple reformatting was not sufficent, but a "fastboot:erase" of the userpartition beforehand did the trick...
